app/src/main/java/mozilla/lockbox/support/TimingSupport.kt

```diff
@@ -17,33 +17,50 @@ import mozilla.lockbox.store.ContextStore
 import mozilla.lockbox.store.SettingStore
 
 @ExperimentalCoroutinesApi
-open class AutoLockSupport(
+open class TimingSupport(
     val settingStore: SettingStore = SettingStore.shared
 ) : ContextStore {
 
     companion object {
-        val shared by lazy { AutoLockSupport() }
+        val shared by lazy { TimingSupport() }
     }
 
     private val compositeDisposable = CompositeDisposable()
     private lateinit var preferences: SharedPreferences
 
-    var lockingSupport: LockingSupport = SystemLockingSupport()
+    var systemTimingSupport: SystemTimingSupport = SystemSystemTimingSupport()
 
     open val shouldLock: Boolean
         get() = lockCurrentlyRequired()
 
+    open val shouldSync: Boolean
+        get() = syncCurrentlyRequired()
+
     override fun injectContext(context: Context) {
         preferences = PreferenceManager.getDefaultSharedPreferences(context)
     }
 
+    private fun syncCurrentlyRequired(): Boolean {
+        val syncTimerDate = preferences.getLong(Constant.Key.syncTimerDate, Long.MAX_VALUE)
+        val currentSystemTime = systemTimingSupport.systemTimeElapsed
+
+        // this will be triggered when the system time is very low due to device restarts
+        val currentTimeTooEarly = (syncTimerDate - currentSystemTime) > Constant.Common.twentyFourHours
+
+        return syncTimerDate < currentSystemTime || currentTimeTooEarly
+    }
+
     open fun storeNextAutoLockTime() {
         settingStore.autoLockTime
             .take(1)
             .subscribe(this::updateNextLockTime)
             .addTo(compositeDisposable)
     }
 
+    open fun storeNextSyncTime() {
+        storeSyncTimerDate(systemTimingSupport.systemTimeElapsed + Constant.Common.twentyFourHours)
```

Comment:
Set `Constant.Common.twentyFourHours` as a property of `TimingSupport`.

e.g. 

```kotlin
private val syncInterval = Constant.Common.twentyFourHours
```

Reply:
I stuck it in the constants with the expectation that it will eventually become a setting that users can configure per #757 .